src/config/tileset.updater.ts

```diff
@@ -99,9 +109,52 @@ export class TileSetUpdater {
       }
 
       // Invalidate the existing tile set as something has changed path
-      if (hasImageryChanged) this.invalidationPaths.push(`/v1/tiles/${tsId}/*`);
+      if (hasImageryChanged) {
+        this.invalidationPaths.push(`/v1/tiles/${tsId}/*`);
+        this.invalidationPaths.push(`/v1/tiles/${tsId.slice(3)}/*`);
+      }
       hasChanges = hasChanges || hasImageryChanged;
+
+      if (IgnoreTileSet.has(i.id)) continue;
+      const prevImages = imgByName.get(i.name) ?? new Map();
+      const prev = prevImages.get(i.projection);
+      if (prev?.uri === i.uri) continue;
+      if (prev != null) throw new Error('Duplicate projection differing uris ' + prev.uri + ' vs ' + i.uri);
+      prevImages.set(i.projection, i);
+      imgByName.set(i.name, prevImages);
+    }
+
+    // Create a tileset based on the imagery name
+    for (const i of imgByName.values()) {
+      if (i.size > 2 || i.size < 1) throw new Error('Duplicate imagery: ' + [...i.values()].map((c) => c.uri).join(', '));
+      const imagery = [...i.values()]; // One or two imagery sets
+
+      // TODO should we sanitize the names here? generally the name is a s3 compliant uri
+      const tsId = 'ts_' + imagery[0].name; // TODO should we remove _RGBA/_RGB from the end of the name
```

Comment:
Do we need to add a projection code inside the tileset id to separate duplicated id for each projection?

Reply:
The imagery object allows storing of projection by EPSG keys `{ 2193: 's3://foo/2193/bar', 3857: 's3://foo/3857/bar', name: 'bar' }`